maven-core/src/main/java/org/apache/maven/lifecycle/internal/DefaultLifecycleExecutionPlanCalculator.java

```diff
@@ -358,6 +368,61 @@ private void finalizeMojoConfiguration( MojoExecution mojoExecution )
         mojoExecution.setConfiguration( finalConfiguration );
     }
 
+    private void checkUnKnownMojoConfigurationParameters( MojoExecution mojoExecution )
+        throws PluginConfigurationException
+    {
+        if ( mojoExecution.getConfiguration() == null || mojoExecution.getConfiguration().getChildCount() == 0 )
+        {
+            return;
+        }
+
+        // first stem get parameter names of current goal
+        Set<String> parametersNames = new HashSet<>();
+        for ( Parameter p : mojoExecution.getMojoDescriptor().getParameters() )
```

Comment:
I unfortunately do not have the time to do a proper review, but I noticed some things at a glance:
- I believe this for loop and the one on line 404 is the same. Have you considered refactoring that to a method? 
- If you do the first point, then this method can all be done in a couple of streams instead of for-loops alternated with streams. Could be a small styling benefit. 
- typo in the comment on line 379 (step*) 
-

Reply:
Thanks for review. I will improvement code.
First I want to confirm that it is good place for such verification.